helix-term/src/ui/statusline.rs

```diff
@@ -433,15 +434,30 @@ where
     F: Fn(&mut RenderContext, String, Option<Style>) + Copy,
 {
     let title = (if context.doc.is_modified() {
-        "[+]"
+        " [+] "
     } else {
-        "   "
+        ""
```

Comment:
What is the reason for the change here? Usually we want status line elements to always take up the same amount of space so they isn't any shifting when they change

Reply:
Should the else case be 5 spaces then? If that is the case, does the readonly statusline element also need to take a consistent amount of space?

Reply:
If the padding here was just added to bring more consistentency my preference would be for doing it in another pr like the one I linked in my other comment

Reply:
Ig this links back to the comment I made, the spacing isn't really consistent at all among the different elements: some of them have spaces only after, some both before and after, some not at all. 
I agree that the file modification should be consistent, but I don't really want the ` [readonly] ` element to take up so much space when it isn't readonly: `          `

Reply:
That's true it could probably shouldn't take up space for non read-only files cause that can't change while editing, but the two messages that can show when a file is read-only should take up the same or similar space

Reply:
What do you mean by the two messages? The second message is really a warning and I think it is fine for the warning to change the size

Reply:
Ah maybe I am mistaken could you share a screenshot of how the warning message looks?

Reply:
![image](https://github.com/helix-editor/helix/assets/87130162/2a87012d-2816-4ff9-bdc8-49079cae1965)